test/Unit/Task/TwigCsFixerTest.php

```diff
@@ -0,0 +1,156 @@
+use GrumPHP\Task\Context\RunContext;
+use GrumPHP\Task\TaskInterface;
+use GrumPHP\Task\TwigCsFixer;
+use GrumPHP\Test\Task\AbstractExternalTaskTestCase;
+
+class TwigCsFixerTest extends AbstractExternalTaskTestCase
+{
+    protected function provideTask(): TaskInterface
+    {
+        return new TwigCsFixer(
+            $this->processBuilder->reveal(),
+            $this->formatter->reveal()
+        );
+    }
+
+    public function provideConfigurableOptions(): iterable
+    {
+        yield 'defaults' => [
+            [],
+            [
+                'triggered_by' => ['twig'],
+                'paths' => ['.'],
+                'level' => 'NOTICE',
+                'config' => null,
+                'report' => 'text',
+                'fix' => false,
+                'no-cache' => false,
+                'debug' => false,
+                'quiet' => false,
+                'version' => false,
+                'ansi' => false,
+                'no-ansi' => false,
+                'no-interaction' => false,
+                'verbose' => false,
+            ]
+        ];
+    }
+
+    public function provideRunContexts(): iterable
+    {
+        yield 'run-context' => [
+            true,
+            $this->mockContext(RunContext::class)
+        ];
+
+        yield 'pre-commit-context' => [
+            true,
+            $this->mockContext(GitPreCommitContext::class)
+        ];
+
+        yield 'other' => [
+            false,
+            $this->mockContext()
+        ];
+    }
+
+    public function provideFailsOnStuff(): iterable
+    {
+        yield 'exitCode1' => [
+            [],
+            $this->mockContext(RunContext::class, ['hello.twig']),
+            function () {
+                $this->mockProcessBuilder('twig-cs-fixer', $process = $this->mockProcess(1));
+                $this->formatter->format($process)->willReturn('nope');
+            },
+            'nope',
+            FixableTaskResult::class
+        ];
+    }
+
+    public function providePassesOnStuff(): iterable
+    {
+        yield 'exitCode0' => [
+            [],
+            $this->mockContext(RunContext::class, ['hello.twig']),
+            function () {
+                $this->mockProcessBuilder('twig-cs-fixer', $this->mockProcess(0));
+            }
+        ];
+    }
+
+    public function provideSkipsOnStuff(): iterable
+    {
+        yield 'no-files' => [
+            [],
+            $this->mockContext(RunContext::class),
+            function () {
+            }
+        ];
+        yield 'no-files-after-triggered-by' => [
+            [],
+            $this->mockContext(RunContext::class, ['notatwigfile.php']),
+            function () {
+            }
+        ];
+        yield 'no-files-in-paths' => [
+            ['paths' => ['src']],
+            $this->mockContext(RunContext::class, ['other/hello.twig']),
+            function () {
+            }
+        ];
+    }
+
+    public function provideExternalTaskRuns(): iterable
```

Comment:
can you add tests for all possible options to make sure the CLI arguments are being built correctly?

Reply:
Sure! Done!